MonsterDice.js

```diff
@@ -1,104 +1,138 @@
-			newblock.css("display", "inline-block");
-
-
-		});
-		
-		$(element).find("span[data-rolltype='damage']").each(function(){
-			newblock = $("<div/>");
-			newblock.css("display", "inline-block");
-			dice_notation=$(this).attr('data-dicenotation');
-			dice_damagetype=$(this).attr('data-rolldamagetype')
-			newblock.append(`<button data-rolldamagetype='${dice_damagetype}' data-exp='${dice_notation}' data-mod='' class='above-roll20'>${dice_notation}</button><button data-rolldamagetype='${dice_damagetype}' data-exp='${dice_notation}' class='above-roll20' data-mod='CRIT'>CRIT</button>`);
-			$(this).replaceWith(newblock);
+	const clickHandler = function(clickEvent) {
+		roll_button_clicked(clickEvent, displayName, creatureAvatar)
+	};
 
-		})
+	const rightClickHandler = function(contextmenuEvent) {
+		roll_button_contextmenu_handler(contextmenuEvent, displayName, creatureAvatar);
+	}
 
+	replace_ability_scores_with_avtt_rollers(target, ".ability-block__stat" ,".ability-block__heading", true)
+	replace_saves_skill_with_avtt_rollers(target, ".mon-stat-block__tidbit", ".mon-stat-block__tidbit-label", ".mon-stat-block__tidbit-data", true )
 
-		// these covers "text parsing" for things that do not support the new DDB roll notations
-		description = $(element).html();
-		description = description.replace(/([\+-][0-9]+) to hit/, "<button data-exp='1d20' data-mod='$1' class='above-roll20'>$1</button><button class='above-roll20' data-exp='2d20kh1' data-mod='$1'>A</button><button class='above-roll20' data-exp='2d20kl1' data-mod='$1'>D</button> to hit");
-		//Spell Slots, or technically anything with 'slot'... might be able to refine the regex a bit better...
-		//TEMPORARILY DISABLED  description = description.replace(/([0-9]) slot/, '<input style="font-size: 14px; width: 40px; appearance: none;" type="number" value="$1">/$1</input> slot');
-		//Actions which go #/Day... again probably could refine the regex a bit better.
-		//TEMPORARILY DISABLED  description = description.replace(/([0-9])\/Day/i, '<input style="font-size: 14px; width: 40px; appearance: none;" type="number" value="$1">/$1</input>Day');
-		//Legendary actions which go #/Day... again probably could refine the regex a bit better.
-		//TEMPORARILY DISABLED description = description.replace(/can take ([0-9]) legendary actions/i, '<input id=legendary_actions style="font-size: 14px; width: 40px; appearance: none;" type="number" value="$1">can take $1 legendary actions</input>');
-		
-		description = description.replaceAll(/\(([0-9]+d[0-9]+( [\+-] [0-9]+)?)\)/g, "(<button data-exp='$1' data-mod='' class='above-roll20'>$1</button><button data-exp='$1' class='above-roll20' data-mod='CRIT'>CRIT</button>)");
-		$(element).html(description);
-		
-		
+	// replace all "to hit" and "damage" rolls
+	$(target).find(".mon-stat-block p").each(function() {
+		let currentElement = $(this)
+		if (currentElement.find(".avtt-roll-button").length == 0) {
+			$(currentElement).find("span").each(function (){

```

Comment:
```suggestion
			$(currentElement).find("span").each(function (){
$(currentElement).find("span[data-dicenotation]").each(function (){
```

This solves the _little death_ hyperlink issue